src/low-level/imap/qresync.c

```diff
@@ -409,7 +409,7 @@ int mailimap_fetch_qresync_vanished(mailimap * session,
     * p_vanished = get_vanished(session);
   }
 
-  if (clist_count(* fetch_result) == 0) {
+  if (* fetch_result != NULL && clist_count(* fetch_result) == 0) {
```

Comment:
if `(* fetch_result == NULL)`, I think we should also do the same thing here.

Could you change the test to the following instead?

```
if (* fetch_result == NULL || clist_count(* fetch_result) == 0) {
  // make the compiler happy by checking value of * fetch_result but it's unlikely to happen.
  error_code = response->rsp_resp_done->rsp_data.rsp_tagged->rsp_cond_state->rsp_type;
} else {
   error_code = MAILIMAP_RESP_COND_STATE_OK;
}
```

Reply:
You right, here.